File: testdata/precision-suite-csharp/tp_xss.cs
// Reflected XSS: an untrusted query parameter is written un-encoded into the
// HTML response via Response.Write — CWE-79. A correct scanner fires TAINT-003.
// The value is reflected verbatim, so "<script>...</script>" executes in the
// victim's browser.
using System.Web;

public class GreetController
{
    // Greet echoes the caller's name straight into the page — the vulnerability.
    public void Greet(HttpRequest Request, HttpResponse Response)
    {
        var name = Request.QueryString["name"];
        Response.Write("<h1>Hello " + name + "</h1>"); // nox-expect: TAINT-003
    }
}

File: testdata/precision-suite-java/tp_xss.java
// Reflected XSS: an unescaped request parameter is written straight into the
// HTML response via the servlet response writer — CWE-79. A correct scanner
// fires TAINT-003. The escaped form lives in clean_safe_output.java.
package com.example.web;

import java.io.IOException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class GreetingServlet {

    // greet reflects untrusted input into the page unescaped — the vulnerability.
    public void greet(HttpServletRequest request, HttpServletResponse response)
            throws IOException {
        String msg = request.getParameter("msg");
        response.getWriter().println("<div>Hello, " + msg + "</div>"); // nox-expect: TAINT-003
    }
}
